public/script.js

```diff
@@ -1,31 +1,34 @@
 document
-  .getElementById("add-task")
-  .addEventListener("click", async function () {
-    const taskInput = document.getElementById("new-task");
-    const taskText = taskInput.value.trim();
+  .getElementById("new-task")
+  .addEventListener("keypress", async function (event) {
+    if (event.key === "Enter") {
+      const taskInput = document.getElementById("new-task");
+      const taskText = taskInput.value.trim();
 
-    if (taskText) {
-      const task = {
-        id: Date.now().toString(),
-        text: taskText,
-        completed: false,
-      };
+      if (taskText) {
+        const task = {
+          id: Date.now().toString(),
+          text: taskText,
+          completed: false,
+        };
 
-      // Send the task to the server
-      const response = await fetch("/tasks", {
-        method: "POST",
-        headers: { "Content-Type": "application/json" },
-        body: JSON.stringify(task),
-      });
+        // Send the task to the server
+        const response = await fetch("/tasks", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify(task),
+        });
 
-      if (response.ok) {
-        const newTask = await response.json();
-        displayTask(newTask); // Function to render the task in the list
-        taskInput.value = "";
+        if (response.ok) {
+          const newTask = await response.json();
+          displayTask(newTask); // Function to render the task in the list
+          taskInput.value = "";
+        }
       }
     }
   });
 
+// Load and display tasks from the server
 async function loadTasks() {
   const response = await fetch("/tasks");
   const tasks = await response.json();
```

Comment:
hey @JKRakuba 
There's no error handling in the loadTasks function. If the fetch request fails or if there's an issue parsing JSON, it would be helpful to catch and log those errors, so you know where the error is.

Reply:
Hi @Tepanyega, Thank you so much for pointing this out, Error handling has been added in Loadtasks() function.